catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/QueryTransformer.java

```diff
@@ -0,0 +1,122 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.workspace.transformer.impl;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.codice.ddf.catalog.ui.metacard.workspace.QueryMetacardImpl;
+import org.codice.ddf.catalog.ui.query.transform.QueryBasicValueTransformer;
+import org.codice.ddf.catalog.ui.query.transform.QuerySortsValueTransformer;
+import org.codice.ddf.catalog.ui.query.transform.QueryValueTransformer;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class QueryTransformer {
+
+  private EndpointUtil endpointUtil;
+
+  private List<QueryValueTransformer> queryValueTransformers;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(QueryTransformer.class);
+
+  public QueryTransformer(EndpointUtil endpointUtil) {
+    this.endpointUtil = endpointUtil;
+
+    // The field could match multiple transformers. The order of these transformers matters.
+    this.queryValueTransformers =
+        Arrays.asList(new QuerySortsValueTransformer(), new QueryBasicValueTransformer());
+  }
+
+  public Map<String, Object> transformMetacardIntoMap(Metacard metacard) {
+    Map<String, Object> map = endpointUtil.getMetacardMap(metacard);
+    List<Map> sorts =
+        Optional.of(map.remove("sorts"))
+            .filter(value -> value instanceof List)
+            .map(List.class::cast)
+            .map(this::mapSorts)
+            .orElse(Collections.emptyList());
+
+    Map<String, Object> filteredMap =
+        map.entrySet()
+            .stream()
+            .filter(entry -> Objects.nonNull(entry.getValue()))
+            .filter(entry -> isBasicType(entry.getValue()))
+            .collect(Collectors.toMap(Entry::getKey, Entry::getValue));
+
+    filteredMap.put("sorts", sorts);
+
+    return filteredMap;
+  }
+
+  private boolean isBasicType(Object value) {
+    return !(value instanceof List) || notEmptyList(value);
+  }
+
+  private boolean notEmptyList(Object value) {
+    return value instanceof List && !((List) value).isEmpty();
+  }
+
+  public Metacard transformJsonIntoMetacard(Map<String, Object> json) {
+    QueryMetacardImpl query = new QueryMetacardImpl();
+
+    json.forEach(
+        (key, value) -> {
+          QueryValueTransformer transformer = getQueryValueTransformer(key, value);
+
+          if (transformer != null) {
+            Attribute attr = transformer.createAttribute(key, value);
+            query.setAttribute(attr);
+          } else {
+            LOGGER.trace(
+                "Appropriate query value transformer could not be found for %s with value %s",
```

Comment:
The logger format string has to use `{}` instead of `%s` (and you don't need to call `toString()` on the arguments, which will save time if trace logging is disabled).